### ISSUE TITLE
Декларативное определение библиотеки на основе структуры проекта

### ISSUE THREAD

**Issue by yukon39** [oscript-library/opm#203]
Взято по аналогии со стандартами разработки Go (https://github.com/golang-standards/project-layout)

Реализовать декларативное описание библиотеки на основании структуры проекта. Например:
* `/cmd` - определяет "ИсполняемыйФайл".
* `/pkg` (`/public`) - Классы и модули расположенные определяют публичные объекты аналогично "ОпределяетКласс"/"ОпределяетМодуль"
* `/bin` - регистрирует расположенные внутри аддины. 
* `/internal` (`/private`) - включает в состав пакета, но не публикует классы/модули
* `/vendor` - включает в состав поставляемые библиотеки (текущий аналог `oscript-modules`)
* `/docs`, `/examples`, `/etc` ... - дополнительные файлы библиотеки (документация, примеры, другое)

Для задания псевдонимов реализовать множественную аннотацию `@Синоним/@Alias`. Например:
`/cmd/main.os`  - имя по умолчанию - создается cmd по имени библиотеки. Для 
```1c
@Alias vrunner
@Alias runner
```
Или реализовать alias.json в папке с файлом:
```json
{
   "main.os": [ "vrunner", "runner" ]
}
``` 
Аналогично, для публичных классов/модулей.

Это позволит:
1. Упростить описание проекта, полностью убрав, или разместив декларации объектов максимально приближенно к их определению
2. Сделает проекты более структурированными, что повысит их читаемость и сопровождаемость

**Comment by nixel2007** [comment]
> Упростить описание проекта, полностью убрав, или разместив декларации объектов максимально приближенно к их определению

надо отметить, что указание ОпределяетКласс и ОпределяетМодуль - полностью опциональное. Если модули и классы лежат по соглашению о структуре каталогов, то загрузчик библиотек из сам загрузит в правильном виде.

**Comment by EvilBeaver** [comment]
Да, у нас и так в целом на основе структуры проекта загрузка ведется. Развивать понятно что можно и дальше. Надо проектировать каждый конкретный кейс: Цель, ЧтобыЧто. @yukon39 возьметесь?

**Comment by yukon39** [comment]
@EvilBeaver К предложенной структуре проекта есть предложения/замечания? Я в рамках ее попробую что-нибудь реализовать.

**Comment by EvilBeaver** [comment]
Замечание одно - это слишком высокоуровневый набросок из которого непонятно есть к нему замечания или нет.
Хотелось бы понять что делает каждая папка, как ее обрабатывает opm, что с ней делает автор проекта и т.п. Как это стыкуется с packegdef... Короче сейчас слишком мало информации для понимания предложения
